core/internal/stream/sender.go

```diff
@@ -161,15 +161,12 @@ type Sender struct {
 	// Keep track of exit record to pass to file stream when the time comes
 	exitRecord *spb.Record
 
+	// Keep track of sync finish record to pass to file stream when the time comes
+	syncFinishRecord *spb.Record
```

Comment:
Add a TODO: to check/handle a case where a finish record is missing in the leveldb log.